### DISCUSSION TITLE
type dictionary parser?

### DISCUSSION THREAD

**Discussion by guykatz** [microstream-one/microstream#653] [Q&A] [answered]
Hi all
can someone point me to a type dictionary file parser in microstream source code?
I assume there is one :)
thanks

**Comment by hg-ms** [comment, marked as answer]
The type-dictionary parser is:
`one.microstream.persistence.types.PersistenceTypeDictionaryParser.Default`
it creates a list of ` PersistenceTypeDictionaryEntry` that is used by the `one.microstream.persistence.types.PersistenceTypeDictionaryBuilder.Default` to create the type-dictionary.

please see `one.microstream.persistence.types.PersistenceTypeDictionaryCompiler.Default.compileTypeDictionary(String)` how it is used.
